### ISSUE TITLE
AMD Ryzen Issue(s) on version 8.1 and 8.2.1

### ISSUE THREAD

**Issue by Include-sys** [xcp-ng/xcp#555]
Specs of my system,

AMD Ryzen 9 3900x
Asus X570-F Gaming (BIOS version 4403, AMD AM4 AGESA V2 PI 1.2.0.7)
G.Skill 3600Mhz CL18 DDR4
MSI GT 1030
2xNVMe Storage (500GB+2TB)

I tried 8.1 and 8.2.1 with a lot of different start combinations(changing cmdline boot option(2048M and etc.), different bios options(disabling PBO, disabling XMP, disabling C-State and etc.)) nothing worked. I kept getting the following error,(First on BIOS, second on UEFI) It wasn't booting.

![IMG20220702122101](https://user-images.githubusercontent.com/10716262/177002322-d1103d14-48aa-4ff8-a3c2-03b7561e04d2.jpg)
![IMG20220702124443](https://user-images.githubusercontent.com/10716262/177002326-f89174ae-98e8-4276-8f82-847c98773a3c.jpg)

Then I tried version 8.0.0 and could install it without changing anything. I wanted to have ext4 and checked official doc troubleshooting before giving it another try and saw iommu=0 option. And It worked, I was configuring the installer but It hung at some point and I decided to try again then I started to get this USB errors,

![IMG20220702144549](https://user-images.githubusercontent.com/10716262/177002470-b5f204f3-f57e-4232-a0f7-2065947242e5.jpg)

I thought It would run without a problem If I installed it once, so I started over every time It hung or showed these errors then I was able to install it. I still have to add iommu=0 option to boot xpg-ng but It's crazy slow, and not usable. It takes 5 mins to get an ssh shell.

So I'm curious If anyone has a solution for this? I really want to use the latest one.

Thanks in advance!

**Comment by Include-sys** [comment]
Here are some additional logs,

[boot.log](https://github.com/xcp-ng/xcp/files/9033184/boot.log)
[crit.log](https://github.com/xcp-ng/xcp/files/9033185/crit.log)
[daemon.log](https://github.com/xcp-ng/xcp/files/9033186/daemon.log)
[hypervisor.log](https://github.com/xcp-ng/xcp/files/9033187/hypervisor.log)
[kern.log](https://github.com/xcp-ng/xcp/files/9033188/kern.log)
[xensource.log](https://github.com/xcp-ng/xcp/files/9033189/xensource.log)

**Comment by olivierlambert** [comment]
Note: related to https://xcp-ng.org/forum/topic/5652/panic-on-cpu-0-xen-bug-at-iommu-init-c-1419

**Comment by geoff313** [comment]
Hi @Include-sys  I had the same problem on two AM4 motherboards (the iommu_init.c error). To fix them I downgrade the BIOS to a version that was pre windows 11 support (but not sure if it was change in the bios to support windows 11 or something else). This worked for an Asus and a MSI board with 8.2.1.

**Comment by ryzen3100** [comment]
Downgrading to pre-win11 BIOS did not fix the issue for me (Asrock B550M Pro4, Ryzen 5800X)

**Comment by konsultaner** [comment]
I have the same issue on:

- AMD Ryzen Threadripper PRO 5975WX
- ASUS Pro WS WRX80E-SAGE
- NVIDIA GeForce RTX 4070 Ti 12GB

any news on this?

**Comment by konsultaner** [comment]
@olivierlambert Thanks for the reply. I went through the forum thread. Is there a beta or later alpha of 8.3 which seem to have worked?

**Comment by olivierlambert** [comment]
No, the patch isn't available yet, but this will come. Stay tuned!

**Comment by martinkeat** [comment]
Confirming - Ryzen 5 3600 here with latest version of xcp-ng (new install, new system) and kernel panic.

**Comment by olivierlambert** [comment]
Can you take a look at https://xcp-ng.org/forum/topic/5652/panic-on-cpu-0-xen-bug-at-iommu-init-c-1419/160?

**Comment by martinkeat** [comment]
Hi @olivierlambert 

Disabled IOMMU and run the current version to no effect.

Ran the Alpha2 version with IOMMU enabled and that failed.

Ran the Alpha2 version with IOMMU disabled and it allowed me in to run the install (which is now happening). Will report back after install and whether I can re-enable IOMMU after install. 

For the record, the system is 4 x Kioxia 480GB SSDs, A320M-K Motherboard just updated with the latest bios PRIME A320M-K BIOS Version 6062 2023/03/21, 32GB of Vengeance LPX DDR4 Ram, AMD Ryzen 5 3600.

**Comment by olivierlambert** [comment]
Have you enabled x2APIC?

**Comment by martinkeat** [comment]
Hi Oliver, yes it was enabled.

Installation ran all the way through with alpha 2 and was able to switch on IOMMU after installation.

**Comment by martinkeat** [comment]
So hopefully when the alpha becomes the LTS version it will all be hunky dorry! Until then, alpha it is.

**Comment by olivierlambert** [comment]
8.3 is already pretty stable, the biggest thing is to carefully check when there's updates. 8.3 won't be an LTS, but it will become an official release at some point. For consumer grade hardware, 8.3 state is probably better than most other Alpha software (there's already many users on it for NUCs and so on)

**Comment by konsultaner** [comment]
@olivierlambert is there a very recent nightly build I can try? Is it good enough for production in a small business environment?

1 Machine, 3 or 4 VMs just filled with web stuff, nothing fancy with GPU or USB or whatever?

**Comment by olivierlambert** [comment]
We don't have nightly builds, latest alpha is 2, an alpha 3 will be out in next weeks, but even alpha 2 is stable. What's less is the potential upgrade path if many things changes, but as long as you have XO backups with it, you shouldn't have any problem. We have dozens of people using it already without any issue. Welcome to the testing club!

**Comment by olivierlambert** [comment]
We have patches for -X CPUs coming in the next weeks, at least in the testing repo or something.

**Comment by stormi** [comment]
@konsultaner Did you try to disable IOMMU, install using 8.2 alpha 2, update the system, and then re-enable IOMMU? This won't solve the known issues, but already works around the kernel panic that was initially the topic of this bug report, if I understood things correctly.

**Comment by konsultaner** [comment]
@stormi I noticed this workarround, but my mainboard doesn't have a setting where I can disable IOMMU.

**Comment by stormi** [comment]
This issue is fixed in the refreshed XCP-ng 8.2.1 installation ISOs. You can test them from https://xcp-ng.org/forum/topic/8026/xcp-ng-8-2-1-refreshed-installation-iso-better-hardware-support and provide feedback. When the testing phase ends, they will become the default installer for XCP-ng 8.2.1.

**Comment by rngadam** [comment]
Still IOMMU crash on AMD Ryzen 7 8845HS w/Radeon 780M Graphics with XCP-ng 8.2.1. Tried both default and "safe" grub option.

Checked that the sha256sum matches the one mentionned in the "refreshed" link.

This is a Beelink SER8 (bios SER8P5C8V29) mini-pc. 

ProxMox v8.2 was successfully installed on same computer.

**Comment by stormi** [comment]
@rngadam Then I think this is a different issue. Maybe related, but not necessarily.
